ptvsd/wrapper.py

```diff
@@ -1532,14 +1536,32 @@ def on_setBreakpoints(self, request, args):
                 self.bp_map.remove(pyd_bpid, vsc_bpid)
 
         cmd = pydevd_comm.CMD_SET_BREAK
-        msgfmt = '{}\t{}\t{}\t{}\tNone\t{}\tNone'
+        msgfmt = '{}\t{}\t{}\t{}\tNone\t{}\t{}\t{}'
         for src_bp in src_bps:
             line = src_bp['line']
             vsc_bpid = self.bp_map.add(
                     lambda vsc_bpid: (path, vsc_bpid))
             self.path_casing.track_file_path_case(path)
+
+            logMessage = src_bp.get('logMessage', '')
+            if len(logMessage) == 0:
+                continue_execution = None
+                condition = src_bp.get('condition', None)
+                expression = None
+            else:
+                continue_execution = False
+                condition = None
+                expressions = re.findall('\{.*?\}', logMessage)
+                if len(expressions) == 0:
+                    expression = 'print("{}")'.format(logMessage.replace('"', '\\"')) # noqa
+                else:
+                    raw_text = reduce(lambda a, b: a.replace(b, '{}'), expressions, logMessage) # noqa
+                    raw_text = raw_text.replace('"', '\\"')
+                    expression_list = ', '.join([s.strip('{').strip('}').strip() for s in expressions]) # noqa
+                    expression = 'print("{}".format({}))'.format(raw_text, expression_list) # noqa
+
             msg = msgfmt.format(vsc_bpid, bp_type, path, line,
-                                src_bp.get('condition', None))
+                                condition, expression, continue_execution)
```

Comment:
I think at this point we should just add CMD_SET_BREAK_JSON and do all of this via JSON? Because we have `suspend_policy` that will eventually come here along with `hit_condition`.

Reply:
Absolutely no idea what that command is, will check that out.

Reply:
I don't mind adding a new command, however what you're suggesting is a change to the structure of the message, i.e. change in the protocol. I'd leave that for a separate PR, its possible such a change might never get approved upstream (as we're introducing different protocols into the library).

Reply:
> I think at this point we should just add CMD_SET_BREAK_JSON and do all of this via JSON? Because we have suspend_policy that will eventually come here along with hit_condition.

the plan is to merge this PR after yours